### PR TITLE
fix comments and return nil directly when err is nil

### PR DESCRIPTION
**Pull request by tanshanshan**
**What this PR does / why we need it**:
1. fix comments 
2. return nil directly when err is nil

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by nikhita** [comment]
/assign @bowei @karataliu
for approval

File: pkg/cloudprovider/providers/gce/token_source.go [hunk 1]
```diff
@@ -71,7 +71,7 @@ func (a *AltTokenSource) Token() (*oauth2.Token, error) {
 	if err != nil {
 		getTokenFailCounter.Inc()
 	}
-	return t, err
+	return t, nil
```
[the remark this answers is not in the document]
Comment (tanshanshan) [reply pkg/cloudprovider/providers/gce/token_source.go]: Done. Thank you.

File: pkg/cloudprovider/providers/azure/azure_managedDiskController.go [hunk 2]
```diff
@@ -174,7 +174,7 @@ func (c *ManagedDiskController) getDisk(resourceGroup, diskName string) (string,
 		return *(*result.DiskProperties).ProvisioningState, *result.ID, nil
 	}
 
-	return "", "", err
+	return "", "", nil
```
Comment (karataliu) [inline pkg/cloudprovider/providers/azure/azure_managedDiskController.go]: This is a case where response does not contain required property, thus should craft an explicit error here. @andyzhangx 

@tanshanshan appreciate the findings, but I'd suggest reverting this file change here.

If you would like to do a fix, I'd suppose changing err to something like
```
fmt.Errorf("Invalid response for getDisk")
```

File: pkg/cloudprovider/providers/azure/azure_managedDiskController.go [hunk 3]
```diff
@@ -174,7 +174,7 @@ func (c *ManagedDiskController) getDisk(resourceGroup, diskName string) (string,
 		return *(*result.DiskProperties).ProvisioningState, *result.ID, nil
 	}
 
-	return "", "", err
+	return "", "", fmt.Errorf("Invalid response for getDisk")
```
Comment (spiffxp) [inline pkg/cloudprovider/providers/azure/azure_managedDiskController.go]: maybe we should say what the result is to aid in troubleshooting?

Comment (tanshanshan) [reply pkg/cloudprovider/providers/azure/azure_managedDiskController.go]: > maybe we should say what the result is to aid in troubleshooting?

so how to describe it 
such as "can not get disk provisionState and diskID "
or
"disk DiskProperties  is nil or provisionState is nil "
Thanks.

Comment (andyzhangx) [reply pkg/cloudprovider/providers/azure/azure_managedDiskController.go]: I think we should just keep the original code, original `err` is nil in Line 177, just keep it.

**Pull request closed without merging**